txpool/fetch.go

```diff
@@ -283,23 +283,31 @@ func (f *Fetch) handleInboundMessage(ctx context.Context, req *sentry.InboundMes
 		}
 	case sentry.MessageId_POOLED_TRANSACTIONS_66, sentry.MessageId_TRANSACTIONS_66:
 		txs := TxSlots{}
-		f.pooledTxsParseCtx.Reject(func(hash []byte) error {
-			known, err := f.pool.IdHashKnown(tx, hash)
-			if err != nil {
-				return err
-			}
-			if known {
-				return ErrRejected
-			}
-			return nil
-		})
 		switch req.Id {
 		case sentry.MessageId_TRANSACTIONS_66:
-			if _, err := ParseTransactions(req.Data, 0, f.pooledTxsParseCtx, &txs); err != nil {
+			if _, err := ParseTransactions(req.Data, 0, f.pooledTxsParseCtx, &txs, func(hash []byte) error {
```

Comment:
Sorry, I'm not following. How making this `checkHash` function a parameter fixes the nil pointer?

Reply:
f.pooledTxsParseCtx - is shared between all methods of `fetch` class. 
when we call `f.pooledTxsParseCtx.Reject(` it does create closure and save it in field of `f.pooledTxsParseCtx` (this closure holds pointer to `tx`)
but if other method of class `fetch` does not call `f.pooledTxsParseCtx.Reject` then it doesn't override existing closure, and doesn't set it to `nil`. Then it will use existing closure which does hold pointer to `tx` which already can be nil. 

also `f.pooledTxsParseCtx` is not thread-safe (it's fixed in `devel`, by method `f.threadSafeParsePooledTxn`) - probably I will add one more PR about it to `stable`.